xdsl/transforms/canonicalization_patterns/arith.py

```diff
@@ -9,15 +9,30 @@
 from xdsl.utils.hints import isa
 
 
-class AddImmediateZero(RewritePattern):
+class AddiIdentityRight(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: arith.AddiOp, rewriter: PatternRewriter) -> None:
-        if (
-            isinstance(op.lhs.owner, arith.ConstantOp)
-            and isinstance(value := op.lhs.owner.value, IntegerAttr)
-            and value.value.data == 0
-        ):
-            rewriter.replace_matched_op([], [op.rhs])
+        if (rhs := const_evaluate_operand(op.rhs)) is None:
+            return
+        if rhs != 0:
+            return
+        rewriter.replace_matched_op((), (op.lhs,))
+
+
+class AddiConstantProp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.AddiOp, rewriter: PatternRewriter):
+        if (lhs := const_evaluate_operand(op.lhs)) is None:
+            return
+        if (rhs := const_evaluate_operand(op.rhs)) is None:
+            # Swap inputs if lhs is constant and rhs is not
+            rewriter.replace_matched_op(arith.AddiOp(op.rhs, op.lhs))
+            return
+
+        assert isinstance(op.result.type, IntegerType | IndexType)
+        rewriter.replace_matched_op(
+            arith.ConstantOp.from_int_and_width(lhs + rhs, op.result.type)
```

Comment:
overflow?

Reply:
I guess we'll crash in the short term and will be able to deal with it then. I just thought I'd raise it now

Reply:
This probably effects the previous PR too

Reply:
I think this is the equivalent part of mlir https://github.com/llvm/llvm-project/blob/9ad09b2930ef2e95bf8772c91f623881d1c14733/mlir/lib/Dialect/Arith/IR/ArithOps.cpp#L325

I can't see any mention of overflow flags or wrapping.